### PR TITLE
Add host view for presenting outside of a view controller

### PR DESCRIPTION
**Pull request by dskuza**
At Pocket, we're investigating new ways to better support YouTube videos. We've been investigating two possible solutions: the official [youtube-ios-player-helper](https://github.com/youtube/youtube-ios-player-helper) framework, or [YouTubePlayerKit](https://github.com/SvenTiigi/YouTubePlayerKit/).

We needed a way to render an appropriate view within a `UICollectionViewCell`, and while one _can_ use the preexisting `YouTubePlayerViewController`, we believe there is more overhead in doing so than adding and reusing a single view to a `UICollectionViewCell`. The other alternative, `YouTubePlayerView`, was not a viable option for us as it required SwiftUI, which we were not using within the context of YouTube support.

Ultimately, we forked the project and added a new class, `YouTubePlayerHostView`, which "hosts" (i.e contains, and hence the name) the underlying `YouTubePlayerWebView` within a `UIView` or `NSView`, depending on your platform, just as we see in both the custom view controller and SwiftUI views.

Usage is similar to that of the view controller as to respect the nature and convenience of its support:

```swift
// Initialize a YouTubePlayerHostView
let youTubePlayerHostView = YouTubePlayerHostView(
    player: "https://youtube.com/watch?v=psL_5RIBqnY"
)

// Example: Access the underlying iFrame API via the `YouTubePlayer` instance
youTubePlayerHostView.player.showStatsForNerds()

// Present YouTubePlayerHostView
addSubview(youTubePlayerHostView)

```

I've done my best to respect the current patterns defined by the project, including that of documentation.

Looking forward to discuss this pull request with you!

### CODE REVIEW

**Comment by SvenTiigi** [comment]
Hi @dskuza,

Thanks for your excellent PR 👍 
Adding a dedicated host view is a great option to avoid the child view controller api especially in a situation where the context/parent-view is a NSView or UIView.

The only thing that I'm not 100% sure is the usage of the `autoResizingMask` which is mainly caused due to personal preferences of mine 😃  
But I think in this scenario the `autoResizingMask`  should do the job.

In summary everything is looking fine!
I'm going to merge the changes into the `develop` branch and release a new version by tomorrow.

**Comment by dskuza** [comment]
@SvenTiigi Sounds good to me! Apologies for targeting the wrong branch. Yay, collaboration!

**Comment by SvenTiigi** [comment]
Hi @dskuza,

Your changes have been merged and are now available with the new version [`1.1.5`](https://github.com/SvenTiigi/YouTubePlayerKit/releases/tag/1.1.5)

I've added one additional change to your PR by updating the naming convention from `YouTubePlayerHostView` to `YouTubePlayerHostingView` (https://github.com/SvenTiigi/YouTubePlayerKit/commit/b23b0fff2c3a05d0c0b50829c9b9a9fdba507d2f)

**Comment by dskuza** [comment]
@SvenTiigi Thanks for the super quick turnaround!
